### PR TITLE
Bitstream writer utility class

### PR DESCRIPTION
**Pull request by mcmasterg**
Note: this needs some cleanup / review in at least the following areas:
-General formatting. I couldn't get make format to do anything (although it does run)
-type 0 packet handling: maybe we should put something in ConfigurationPacket to make this easier to add them? If we don't want InitWithWords to make a type 0 packet, maybe something like ConfigurationPacket::type0 static member, a function, or something of that sort

### CODE REVIEW

**Comment by mithro** [comment]
@kc8apf - Please take a look at this for John.

File: lib/include/prjxray/xilinx/xc7series/bitstream_writer.h [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		absl::optional<packet_iterator> op_itr_packet;
+	};
+
+	BitstreamWriter(const packets_t& packets);
+
+	iterator begin();
+	iterator end();
+
+       private:
+	static header_t header;
```
Comment (kc8apf) [inline lib/include/prjxray/xilinx/xc7series/bitstream_writer.h]: Member variables get suffixes with an underscore.

File: lib/xilinx/xc7series/bitstream_writer.cc [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+/**************************************************
+ *
+ *************************************************/
+
+BitstreamWriter::packet_iterator BitstreamWriter::iterator::packet_begin() {
+	// itr_packets = packets.begin();
+	const ConfigurationPacket& packet = *itr_packets;
+
+	return BitstreamWriter::packet_iterator(
+	    &packet, BitstreamWriter::packet_iterator::STATE_BEGIN,
```
Comment (kc8apf) [inline lib/xilinx/xc7series/bitstream_writer.cc]: How does STATE_BEGIN get changed to STATE_HEADER?  I'm not seeing how this works.

Comment (mcmasterg) [reply lib/xilinx/xc7series/bitstream_writer.cc]: STATE_BEGIN = 1,
			STATE_HEADER = 1,
I can eliminate STATE_BEGIN if that is confusing

File: lib/include/prjxray/xilinx/xc7series/bitstream_writer.h [hunk 3]
```diff
@@ -57,11 +57,11 @@ class BitstreamWriter {
 
 		// Data iterators
 		// First over the fixed header, then the configuration data
-		state_t state;
+		state_t _state;
```
Comment (kc8apf) [inline lib/include/prjxray/xilinx/xc7series/bitstream_writer.h]: Underscore should be a suffix, not a prefix.

**Pull request merged**